lib/puppet/type/service.rb

```diff
@@ -60,6 +63,16 @@ module Puppet
         provider.manual_start
       end
 
+      # This only makes sense on systemd systems. Otherwise, it just defaults
+      # to disable.
+      newvalue(:mask, :event => :service_disabled) do
```

Comment:
Sorry about the merge, revert, and new PR mess, that was my fault.

In the previous PR, we were talking about whether it makes sense to write `ensure => mask` in cases where the provider isn't maskable. Should puppet reject that as an error or interpret that to mean `disable`? The former is easier for manifest authors, because you don't have to know *a priori* if the service provider is maskable or not. The latter providers cleaner semantics about the user's intent -- if you requested `mask` but the system doesn't support it, we fail.

I don't have a good sense for whether the latter case is important. Perhaps maskable and disabled are so similar the distinction isn't meaningful?

For comparison, puppet's package type confines the `ensure` property based on `purgeable` and `holdable`: features: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/type/package.rb#L89-L95.

/cc @nfagerlund @kylog @ScottGarman @adrienthebo